### DISCUSSION TITLE
Emote Modifier Documentation and Prerendering

### DISCUSSION THREAD

**Discussion by Suyooo** [night/betterttv#6757] [Ideas]
Currently, the BTTV API docs are missing information regarding modifier emotes, neither the available types of effects nor the `modifier` property in the response for `/cached/emotes/global`.  
It would be nice to have all available modifiers and their effects listed, to make adding them to 3rd party applications like chat displays easier. A page with the modifier names and effect previews, maybe also the CSS rule used by the extension as a reference for implementation, would be very useful.

My main question though is, are there any plans to offer an API endpoint to prerender emotes with modifiers? It would make it much easier to implement these. For one, while I do have CSS available for the chat display, that is not always the case in other projects, and modifier effects might be difficult to implement in some engines. Mainly though, it's the issue of additional modifiers being added later.  
Right now, 3rd party developers must keep track of new modifiers, figure out an implementation for the effects and push an update for their application every time one is added. It is already possible to get an up-to-date list of modifier names to parse from the global emotes endpoint, so if it was possible to send the parsed modifiers as part of the emote image request and getting the modified emote prerendered from the server, it would not only allow effects to look consistent across all applications, but make it much simpler for developers who are interested in implementing the BTTV API to support emote modifiers.

**Comment by night** [comment]
Thanks for the feedback. We'll have to look into a better way to expose transformations in the API. In the meantime, you could filter unsupported modifiers until they are supported in your app.

As for providing an endpoint that applies the transformation, we have generally avoided image transformation endpoints as they are not really cost effective to operate at scale.
